src/services/JWTService.ts

```diff
@@ -19,38 +20,34 @@ class JWTService {
       throw new AuthorizationError(ERRORMESSAGES.AZURE_CONFIGURATION_NOT_VALID);
     }
 
-    if (!this.isAtLeastOneRoleValid(decodedToken)) {
-      throw new AuthorizationError("Invalid roles");
-    }
+    if (!this.isAtLeastOneRoleValid(decodedToken)) { throw new AuthorizationError("Invalid roles"); }
 
     const endpoint = config.azure.jwk_endpoint.replace(":tennant", config.azure.tennant);
-    return this.fetchJWK(endpoint, decodedToken.header.kid)
-      .then((x5c: string) => {
-        const issuer = config.azure.issuer.replace(":tennant", config.azure.tennant);
-        const certificate = `-----BEGIN CERTIFICATE-----\n${x5c}\n-----END CERTIFICATE-----`;
+    const {header: {kid}} = decodedToken;
+    const {aud: audience} = decodedToken.payload;
 
-        return JWT.verify(token, certificate, {audience: decodedToken.payload.aud, issuer, algorithms: ["RS256"]});
-        // return JWT.verify(token, certificate, { audience: config.azure.appId, issuer, algorithms: ["RS256"] });
-      });
+    try {
+      const x5c = await this.fetchJWK(endpoint, kid);
+      const issuer = config.azure.issuer.replace(":tennant", config.azure.tennant);
+      const certificate = `-----BEGIN CERTIFICATE-----\n${x5c}\n-----END CERTIFICATE-----`;
+
+      return JWT.verify(token, certificate, {audience, issuer, algorithms: ["RS256"]});
+    } catch (e) {
+      console.error("failed getting the certificate");
+      console.log(JSON.stringify(e, null, 2));
+      throw(e);
+    }
   }
 
   /**
    * Internal function used to determine if the user has a valid role. Not directly exposed
    * @param decodedToken
    */
   public isAtLeastOneRoleValid(decodedToken: any): boolean {
-    let isAtLeastOneRoleValid = false;
-    const allowedRoles = [ALLOWEDROLES.CVSFullAccess, ALLOWEDROLES.CVSPsvTester, ALLOWEDROLES.CVSHgvTester, ALLOWEDROLES.CVSAdrTester, ALLOWEDROLES.CVSTirTester, ALLOWEDROLES.CVSVTMAdmin];
-    const rolesOnToken = decodedToken.payload.roles;
-    if (!rolesOnToken) {
-      return false;
-    }
-    allowedRoles.forEach((allowedRole) => {
-      if (rolesOnToken.includes(allowedRole)) {
-        isAtLeastOneRoleValid = true;
-      }
-    });
-    return isAtLeastOneRoleValid;
+    const {payload: {roles}} = decodedToken;
+
+    return roles ? ALLOWED_CVS_ROLES.some((r) => roles.includes(r))
```

Comment:
can be shortened `!!(roles && ALLOWED_CVS_ROLES.some((r) => roles.includes(r))` to get remove the ternary unless a "false" role is passed.